### PR TITLE
Update artifact success message with a view btn 

### PR DESCRIPTION
**Pull request by Cerchie**
## Summary of Changes

begins to address #2661 

> [!NOTE]
> I would _love_ to follow on to this with a PR that highlights the artifact itself instead of just opening the artifacts view on button click when you've got UDFs open. If you think about it that's an edge case. However, the real value is in some test/function refactoring in this PR that added some diff and I didn't want to add more overhead since to add a highlight of a specific artifact[ we'd need to track the tree item like we do with schemas](https://github.com/confluentinc/vscode/blob/e08586617f28692a0a59f329b2b3d7eb63c1f129/src/viewProviders/schemas.ts#L381). 

### Click-testing instructions

1. Open the extension dev host, sign in to ccloud. 

2. Start the upload artifact flow. Once you click the final step, "upload artifact", make like a 🥷 and quickly switch ⚡  to UDF view. If you're already in the Artifact view, nothing much will happen. :( Once we get this merged with the new describe blocks I can start on tracking tree views so I can do this + highlight the new artifact. 
 
3. Once upload is completed, you should see a "View Artifacts" button. Click it, confirm that the view switches from UDFs back to artifacts. 

### Optional: Any additional details or context that should be provided?

I added some `describe` blocks to the `flinkArtifacts` test. Gulp was reading the coverage out as **0** due to the bad describe blocks (read: no describe blocks other than top-level) earlier, so I added some and now we're up to  **60%+** according to gulp. 

Unfortunately, because we're changing this up, SQ counts old unrelated code as part of its review:
Screenshot 2025-10-14 at 2 05 08 PM

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

#### Tests

- [x] Added new
- [x] Updated existing
- [ ] Deleted existing

#### Release notes

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md)?

### CODE REVIEW

File: src/commands/flinkArtifacts.test.ts [hunk 1]
```diff
@@ -40,149 +44,189 @@ describe("flinkArtifacts", () => {
...
-    await uploadArtifactCommand();
-
-    sinon.assert.calledOnce(handleUploadStub);
-    sinon.assert.calledWithExactly(handleUploadStub, mockParams, mockPresignedUrlResponse);
-
-    sinon.assert.calledOnce(createArtifactStub);
-    sinon.assert.calledWithExactly(createArtifactStub, mockParams, mockUploadId);
+  describe("handleWithProgressForUploadArtifact", () => {
+    it("should report progress correctly through all steps", async () => {
+      const mockCreateResponse = {
```
Comment (Cerchie) [inline src/commands/flinkArtifacts.test.ts]: I almost want to put this and `mockProgress` in the top-level describe by default now... but it's the first instance

File: src/commands/flinkArtifacts.ts [hunk 2]
```diff
@@ -14,13 +14,14 @@ import { CCloudKafkaCluster } from "../models/kafkaCluster";
 import {
   showErrorNotificationWithButtons,
   showInfoNotificationWithButtons,
+  showWarningNotificationWithButtons,
 } from "../notifications";
 import { getSidecar } from "../sidecar";
 import { logUsage, UserEvent } from "../telemetry/events";
 import { FlinkDatabaseViewProviderMode } from "../viewProviders/multiViewDelegates/constants";
 import { artifactUploadQuickPickForm } from "./utils/artifactUploadForm";
-import { detectClassesAndRegisterUDFs } from "./utils/udfRegistration";
```
[the remark this answers is not in the document]
Comment (Cerchie) [reply src/commands/flinkArtifacts.ts]: gah, was looking at logUsage only in the merge editor for some reason, now I see that registration code, putting back in

File: src/commands/flinkArtifacts.ts [hunk 3]
```diff
@@ -40,6 +42,39 @@ import {
...
+      status: "succeeded",
+      kind: "CloudProviderUpload",
+      cloud: params?.cloud,
+      region: params?.region,
+    });
+    void showInfoNotificationWithButtons(
+      `Artifact "${response.display_name}" uploaded successfully to Confluent Cloud.`,
+      {
+        "Register UDFs": async () => {
+          if (params?.selectedFile) {
```
Comment (noeldevelops) [inline src/commands/flinkArtifacts.ts]: see above - params should not be allowed/expected to be `undefined`

**Comment by Cerchie** [comment]
closing: deferring discussion on withProgress to get more input, will refactor the test file in a follow-on
